rust/src/composepost.rs

```diff
@@ -172,18 +172,19 @@ fn postprocess_presets(rootfs_dfd: &openat::Dir) -> Result<()> {
     Ok(())
 }
 
-// We keep hitting issues with the ostree-remount preset not being
-// enabled; let's just do this rather than trying to propagate the
-// preset everywhere.
+/// Ensure our own `%_dbpath` macro exists in the tree.
 fn postprocess_rpm_macro(rootfs_dfd: &openat::Dir) -> Result<()> {
     let rpm_macros_dir = "usr/lib/rpm/macros.d";
     rootfs_dfd.ensure_dir_all(rpm_macros_dir, 0o755)?;
     let rpm_macros_dfd = rootfs_dfd.sub_dir(rpm_macros_dir)?;
-    rpm_macros_dfd.write_file_with("macros.rpm-ostree", 0o644, |w| -> Result<()> {
+    let macro_filename = "macros.rpm-ostree";
+    rpm_macros_dfd.write_file_with_sync(&macro_filename, 0o644, |w| -> Result<()> {
         w.write_all(b"%_dbpath /")?;
         w.write_all(RPMOSTREE_RPMDB_LOCATION.as_bytes())?;
+        w.write(b"\n")?;
         Ok(())
     })?;
+    rpm_macros_dfd.set_mode(macro_filename, 0o644)?;
```

Comment:
Ohh wait, you're changing the mode of the *directory* - why?  Wait, aren't you making it not-executable?

Reply:
This the mode not subject to umasking, mostly just because I added a unit-test later on it and I didn't want to consider umask effects there. I don't think this has much effect in real workload, so it can be dropped if you prefer.

Reply:
Ah ok.  Hmm well....I think this gets back to we should be canonicalizing all file modes in ostree commit *or* we add "umask-ignoring" filesystem APIs so we don't repeat ourself here.

Reply:
But OK as is!